src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -575,15 +577,197 @@ public String getName() {
 		public String docs() {
 			return "void {string id, int maxWait} Waits for the thread with the given id to exit. By default, we wait"
 					+ " potentially forever, but if maxWait is specified, we will only wait that many milliseconds."
-					+ " (Sending 0 for this value causes an infinite wait.) If the timeout occurs, an"
-					+ " InterruptedException is thrown. If the id is unknown, an IllegalArgumentException is thrown.";
+					+ " (Sending 0 for this value causes an infinite wait.) If the timeout occurs or thread interrupted, an"
+					+ " InterruptedException is thrown.";
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+	}
+
+	@api
+	public static class x_thread_is_alive extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			String id = args[0].val();
+			Thread th = THREAD_ID_MAP.get(id);
+			if(th == null) {
+				return CBoolean.FALSE;
+			}
+			return CBoolean.get(th.isAlive());
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_thread_is_alive";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "bool {mixed thread} Tests if this thread is alive. A thread is alive if it has been started and has not yet died.";
+		}
+	}
+
+	@api
+	@seealso({x_is_interrupted.class})
+	public static class x_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
```

Comment:
It would be nice to have `x_interrupt()` without arguments interrupt the current thread.